src/Collection.php

```diff
@@ -1136,4 +1214,89 @@ protected function _update($filter, $update, $options)
         $bulk->update($filter, $update, $options);
         return $this->manager->executeBulkWrite($this->ns, $bulk, $this->wc);
     }
+
+    /**
+     * Create one or more indexes for the collection using the createIndexes
+     * command.
+     *
+     * @param Server $server
+     * @param array  $indexes
+     */
+    private function createIndexesCommand(Server $server, array $indexes)
+    {
+        $command = new Command(array(
+            'createIndexes' => $this->collname,
+            'indexes' => $indexes,
+        ));
+        $server->executeCommand($this->dbname, $command);
+    }
+
+    /**
+     * Create one or more indexes for the collection by inserting into the
+     * "system.indexes" collection (MongoDB <2.6).
+     *
+     * @param Server $server
+     * @param array  $indexes
+     */
+    private function createIndexesLegacy(Server $server, array $indexes)
+    {
+        $bulk = new BulkWrite(true);
+
+        foreach ($indexes as $index) {
+            $bulk->insert($index);
+        }
+
+        // TODO: This inherits the client's write concern. Should we enforce {w:1} here? (see: SPEC-195)
+        $server->executeBulkWrite($this->dbname . '.system.indexes', $bulk);
+    }
+
+    /**
+     * Generates an index name from its key specification.
+     *
+     * @param object $key
+     * @return string
+     */
+    private function generateIndexName(stdClass $key)
+    {
+        $name = '';
+
+        foreach ($key as $field => $type) {
+            $name .= ($name != '' ? '_' : '') . $field . '_' . $type;
```

Comment:
It's possible for `$type` to be a non-scalar here, since we don't validate the `$key` object's contents earlier. We should do so in `createIndexes()`.